Microsoft.Toolkit.Uwp.UI.Controls/AdMediator/UwpAdMediator.xaml.cs

```diff
@@ -0,0 +1,377 @@
+        public static readonly DependencyProperty AdDuplexAppKeyProperty =
+            DependencyProperty.Register("AdDuplexAppKey", typeof(string), typeof(UwpAdMediator), new PropertyMetadata(null));
+
+        public string AdDuplexAdUnit
+        {
+            get { return (string)GetValue(AdDuplexAdUnitProperty); }
+            set { SetValue(AdDuplexAdUnitProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for AdDuplexAdUnit.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty AdDuplexAdUnitProperty =
+            DependencyProperty.Register("AdDuplexAdUnit", typeof(string), typeof(UwpAdMediator), new PropertyMetadata(null));
+
+        public int AdRefreshSeconds
+        {
+            get { return (int)GetValue(AdRefreshSecondsProperty); }
+            set { SetValue(AdRefreshSecondsProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for AdRefreshSeconds.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty AdRefreshSecondsProperty =
+            DependencyProperty.Register("AdRefreshSeconds", typeof(int), typeof(int), new PropertyMetadata(35));
+
+        public int MaxErrorsPerRefresh
+        {
+            get { return (int)GetValue(MaxErrorsPerRefreshProperty); }
+            set { SetValue(MaxErrorsPerRefreshProperty, value); }
+        }
+
+        // Using a DependencyProperty as the backing store for MaxErrorsPerRefresh.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty MaxErrorsPerRefreshProperty =
+            DependencyProperty.Register("MaxErrorsPerRefresh", typeof(int), typeof(UwpAdMediator), new PropertyMetadata(3));
+
+        // Dispatch timer to fire at each ad refresh interval.
+        private DispatcherTimer myAdRefreshTimer = new DispatcherTimer();
+
+        // Global variables used for mediation decisions.
+        private Random randomGenerator = new Random();
+        private int errorCountCurrentRefresh = 0;  // Prevents infinite redirects.
+        private int adDuplexWeight = 0;            // Will be set by GetAdDuplexWeight().
+
+        // Microsoft and AdDuplex controls for banner ads.
+        private AdControl myMicrosoftBanner = null;
+        private AdDuplex.AdControl myAdDuplexBanner = null;
+
+        private string MicrosoftAppId => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileApplicationId : this.DesktopApplicationId;
+
+        private string MicrosoftAdUnit => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileAdUnit : this.DesktopAdUnit;
+
+        private int AdWidth => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileAdWidth : this.DesktopAdWidth;
+
+        private int AdHeight => (AnalyticsInfo.VersionInfo.DeviceFamily == "Windows.Mobile") ? this.MobileAdHeight : this.DesktopAdHeight;
```

Comment:
This doesn't work for Continuum. The recommended way is to use physical screen size (<7" I believe)